app/commands/user_command.rb

```diff
@@ -39,14 +39,16 @@ def efficiently_delete_relations(time = Time.now)
     Playlist.joins(:assets).where(assets: { user_id: user.id })
             .update_all(['tracks_count = tracks_count - 1, playlists.updated_at = ?', time])
     Track.joins(:asset).where(assets: { user_id: user.id }).update_all(deleted_at: time)
-    Comment.joins("INNER JOIN assets ON commentable_type = 'Asset' AND commentable_id = assets.id")
-           .joins('INNER JOIN users ON assets.user_id = users.id').where('users.id = ?', user.id).update_all(deleted_at: time)
-
     user.assets.update_all(deleted_at: time)
 
+    # comments given
+    user.given_comments.update_all(deleted_at: time)
```

Comment:
@sudara correct me if I'm wrong, but we might have wanted to leave any comments behind when a user is deleted. A quick check on my data showed a whole lot of comments without commenter. Plenty of them are not-spam. Plent-ier yet - spam.

I would like to suggest that we consider if we want to tighten the rules for what a guest user can do. Does it really matter to you as a musician to get an anonymous comment on your music?

Let's say we choose to leave comments behind when user is deleted, but let's NOT leave spam ones at least. 

Thoughts?

Reply:
> correct me if I'm wrong, but we might have wanted to leave any comments behind when a user is deleted

Historically we've kept true to our promise that we delete ALL the content you create when you delete your account. This has a secondary side effect of removing comments on other people's tracks where you might have said nice things, but I believe it's the cleanest solution currently.

> Does it really matter to you as a musician to get an anonymous comment on your music?

Yes, most family members or non-musician friends of musicians use guest comments quite regularly.